src/atlas-orca/meshgenerator/LocalOrcaGrid.cc

```diff
@@ -240,7 +240,7 @@ PointXY LocalOrcaGrid::normalised_grid_xy( idx_t ix, idx_t iy ) const {
       west = lon00_ - 20.;
   }
 
-  if ( ij.i < nx_orca_ / 2 ) {
+  if ( ix + ix_orca_min_ < orca_.nx() / 2 ) {
```

Comment:
I don't think I understand this one (sorry its been a while since we made this change and I am trying to remind myself of the context).

Firstly its different to the same limit in `normalised_grid_master_lonlat` where previously these two cases were the same. whats the reason for that? Secondly, this change uses `ix` rather than the master coordinate. I think in the original code this used the master coordinate:
https://github.com/ecmwf/atlas-orca/blob/d9ff4d39ff3e7a420548a207fef60969c2bbf4f4/src/atlas-orca/meshgenerator/OrcaMeshGenerator.cc#L396

Reply:
Does the above hold with halo > 1 and mpi > 2? Sorry I am having a lot of trouble building atlas-orca at the moment at the Met Office (it will get easier once we all move to atlas 0.38.1) so it is tricky for me to check for myself right now.
 
I believe that the concept of the "master ij" changes its meaning a little when we introduce halos. We have been treating it as the ij point corresponding to the "master index" where this has the specific meaning of the index to the unique non-orca-halo point in the grid for that data. We do not use the word "master" for any other meaning.... at least that was the intention! :)

The point you are referring to is what we have called the `global_ij` point. Just the local ix, iy translated to global ij space with no wrapping.

I am not sure what it is we want, but I am very keen for us to use labelled coordinate systems. When doing this change I believe we counted about 7 coordinate systems in play - some of them reduce to others of them, but it is very confusing and the differences are often slight! This was my go at listing them off:

code | Indexing scheme for indexing 2D points into 1D arrays (i,j   meanings depend on the schema) | Methods/arrays using this scheme
-- | -- | --
Global regular | 0 indexing on the global regular grid (for indexing 1D array   regular grid objects) [j*nx_glb + i] | Atlas regular grid object, distribution.partition indexing
Local regular | 0   indexing on the local regular grid (for indexing 1D array properties of this   object) [j*nx + i] | is_node, is_ghost, partition array
Haloed local regular | Haloed indexing on local regular grid (starting from (-halo,   -halo)) | Surrounding rectangle
Global orca | 0   indexing on the global orca grid (for indexing 1D array OrcaGrid objects)   [j*orcagrid.nx() + i] | OrcaGrid   internal indexing
Folded global orca | Orcafold-Haloed indexing on the global orca grid (starting from   (-orca_halo_w, -orca_halo_s), used for indexing things via PeriodicIndex   function) [offset = -j_min *    orca_nx_halo – i_min; offset + j*orca_nx_halo + i] | OrcaGrid::PeriodicIndex to wrap around the orca grid and return   correct node.
Haloed local orca | Orcafold-haloed   indexing on the local orca grid (including the orca halos) [offset = -j_min   *  orca_nx_halo – i_min; offset +   j*orca_nx_halo + i] | Local Orca   Grid object when there are no mpi halos
Haloed   Local   Orca | Orcafold-and-mpi-haloed indexing on the local orca grid   (including both the orcafold points and the mpi haloed points) [offset =   -j_min *  orca_nx_halo – i_min; offset   + j*orca_nx_halo + i] | Local Orca Grid object when there are mpi halos
Local orca | 0   indexing on the local orca grid [j*nx_orca + i] | Local   Orcagrid is_node, is_ghost, partition array

Reply:
Its interesting how the water/land mask has become messed up too in the version we had. Honestly I kind of prefer the second version because it makes the cells in the north fold visible - it really does have that wacky shape. I can see it isn't playing nicely with visualisation though.